File: connectors/mainapp/src/main/java/com/taskadapter/config/StoredConnectorConfig.java
package com.taskadapter.config;

/**
 * Stored connector configuration. Just a plain data and nothing else.
 */
public final class StoredConnectorConfig {
    private final String connectorTypeId;
    private String serializedConfig;

    public StoredConnectorConfig(String connectorTypeId,
            String serializedConfig) {
        this.connectorTypeId = connectorTypeId;
        this.serializedConfig = serializedConfig;
    }

    public String getSerializedConfig() {
        return serializedConfig;
    }

    public void setSerializedConfig(String serializedConfig) {
        this.serializedConfig = serializedConfig;
    }

    public String getConnectorTypeId() {
        return connectorTypeId;
    }

}

File: webui/src/main/scala/com/taskadapter/config/StoredConnectorConfig.scala
package com.taskadapter.config

import com.taskadapter.web.uiapi.SetupId

/**
  * Stored connector configuration. Just a plain data and nothing else.
  */
case class StoredConnectorConfig(connectorTypeId: String, connectorSavedSetupId: SetupId, serializedConfig: String)